src/plugins/hexnumber/hexnumber.c

```diff
@@ -0,0 +1,439 @@
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	const size_t length = (size_t) result + 1;
+	char * hexValue = elektraMalloc (length);
+	if (!hexValue)
+	{
+		ELEKTRA_MALLOC_ERROR (parentKey, length);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	if (snprintf (hexValue, length, "0x%llx", value) < 0)
+	{
+		ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_INVALID_FORMAT, parentKey, "Unable to convert '%s' into hexadecimal", decValue);
+		elektraFree (hexValue);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	// set hex string in key and unset internal metadata
+	keySetString (key, hexValue);
+	keySetMeta (key, ELEKTRA_HEXNUMBER_META_KEY, "0");
+
+
+	elektraFree (hexValue);
+
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+/**
+ * Checks whether a given Key's value is a hexadecimal string.
+ *
+ * @param key The Key that should be checked.
+ *
+ * @retval #true if the Key's value starts with 0x
+ * @retval #false otherwise
+ */
+static bool isHexString (const Key * key)
+{
+	return strncasecmp (keyString (key), "0x", 2) == 0;
+}
+
+/**
+ * Checks whether a given Key is specified to have a hexadecimal base.
+ *
+ * @param key The Key that should be checked.
+ *
+ * @retval #true if the Key's metadata contains the key /unit/base and its value is hex
+ * @retval #false otherwise
+ */
+static bool isHexUnitBase (const Key * key)
+{
+	const Key * unitBaseMeta = keyGetMeta (key, "unit/base");
+	return strcmp (keyString (unitBaseMeta), "hex") == 0;
+}
+
+/**
+ * Checks whether a given key has one of the given types.
+ *
+ * @param key The Key that should be checked.
+ * @param types A NULL terminated list of strings containing possible values for the /type metadata.
+ *
+ * @retval #true if the Key's type metadata is contained in the types list
+ * @retval #false otherwies
+ */
+static bool hasType (const Key * key, char ** types)
+{
+	const Key * typeMeta = keyGetMeta (key, "type");
+	if (!typeMeta)
+	{
+		return false;
+	}
+
+	const char * type = keyString (typeMeta);
+	for (int i = 0; types[i] != NULL; ++i)
+	{
+		if (strcmp (types[i], type) == 0)
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Checks whether a given Key's type metadata is #ELEKTRA_HEXNUMBER_META_TYPE.
+ *
+ * @param key The Key that should be checked.
+ *
+ * @retval #true if the Key's type metadata is #ELEKTRA_HEXNUMBER_META_TYPE
+ * @retval #false otherwise
+ */
+static bool hasHexType (const Key * key)
+{
+	const Key * typeMeta = keyGetMeta (key, ELEKTRA_HEXNUMBER_META_KEY);
+	return typeMeta && strcmp (keyString (typeMeta), "1") == 0;
+}
+
+/**
+ * copied from boolean/boolean.c
+ */
+static void strToArray (Key * key, char *** array)
+{
+	int count = 1;
+	const char * values = keyString (key);
+	char * ptr = (char *) values;
+	while (*ptr)
+	{
+		if (*ptr == ';') ++count;
+		++ptr;
+	}
+	*array = elektraCalloc ((count + 1) * sizeof (char *));
+	char * localString = strdup (values);
```

Comment:
please use elektraStrDup (for everything that allocates memory, there is an elektra* method)